client/src/main/java/io/pravega/client/stream/impl/StreamCutImpl.java

```diff
@@ -184,6 +186,37 @@ private void write00(StreamCutInternal cut, RevisionDataOutput revisionDataOutpu
         }
     }
 
+    // StreamCut serializer for version 1.
+    public static class StreamCutSerializer01 extends StreamCutSerializer {
+
+        @Override
+        protected byte getWriteVersion() {
+            return 1;
+        }
+
+        @Override
+        protected void declareVersions() {
+            super.declareVersions();
+            version(1).revision(0, this::write10, this::read10);
+        }
+
+        private void read10(RevisionDataInput revisionDataInput, StreamCutBuilder builder) throws IOException {
```

Comment:
We are adding a new version, which enables us to deserialize a data structure with an older format, but we are not doing anything with it. In particular, if we get a stream cut with `00` format, then we should be executing the previous logic or perhaps implementing a reasonable workaround.

Also, it might be a more elegant solution to extend `RevisionDataOutput` to include something like `writeCompactSignedLong` and `writeCompactSignedLong` that uses a bit for the signal rather than use stock `java.io`.

Reply:
Yes, the older logic is invoked based on the version number. the Junit `io.pravega.client.stream.StreamCutTest#testStreamCutSerializationCompatabilityV0` validates this behaviour.

Reply:
Updated the code to use the same VersionedSerializer, `io.pravega.client.stream.impl.StreamCutImpl.StreamCutSerializer` for both the versions.

Reply:
@shrids you have not responded to this:

> Also, it might be a more elegant solution to extend RevisionDataOutput to include something like writeCompactSignedLong and writeCompactSignedLong that uses a bit for the signal rather than use stock java.io.

In the case it is not clear what I'm suggesting, I'd rather see something like this in `write10`:

```
revisionDataOutput.writeMap(map, (out, s) -> out.writeCompactLong(s.getSegmentId()),
                                 (out, offset) -> out.writeSignedCompactLong(offset));
```

Reply:
Pull request https://github.com/pravega/pravega/pull/3782/files is addressing this. Will update the PR once the changes are in.

Reply:
fixed it.